sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -206,22 +206,22 @@ private[sql] case class PreWriteCheck(conf: SQLConf, catalog: SessionCatalog)
         // The relation in l is not an InsertableRelation.
         failAnalysis(s"$l does not allow insertion.")
 
-      case c: CreateTableUsingAsSelect =>
+      case CreateTable(tableDesc, mode, Some(query)) =>
```

Comment:
Previously, this is only applicable to Data Source tables. After this change, this is also applicable to Create Hive Table As Select. Thus, some validation might not be right to Hive tables. We have to be careful to check them one by one.